src/functions.php

```diff
@@ -392,3 +392,23 @@ function _idn_uri_convert(UriInterface $uri, $options = 0)
 
     return $uri;
 }
+
+/**
+ * @param string $name
+ *
+ * @return string|null
+ *
+ * @internal
+ */
+function _getenv($name)
+{
+    if (isset($_SERVER[$name])) {
+        return (string) $_SERVER[$name];
+    }
+
+    if (PHP_SAPI === 'cli' && ($value = getenv($name) !== false)) {
```

Comment:
I like the concept but please help me understand this.. if someone has been using putenv (let's say on a boostrap file of his own) wont this break for him?

Reply:
https://3v4l.org/l4h7T

Reply:
Yes. That is intentional. Because calling `putenv` is what is not safe to do. This is the break we have to make in order to fix things. People should be loading stuff into the server super global, and not using `putenv` if they are bootstrapping something to run in a multithreaded environment, such as php-fpm.